mods/ctf/ctf_map/maps_catalog.lua

```diff
@@ -133,25 +158,24 @@ minetest.register_chatcommand("maps", {
 		end
 
 		local player = minetest.get_player_by_name(name)
-		local idx
 
 		-- If arg. supplied, set idx to index of the matching map name
 		-- or path. Else, set to indices[name] or index of current map
 		if param then
-			idx = ctf_map.get_idx_and_map(param)
-		else
-			idx = (player and indices[name]) or ctf_map.map and ctf_map.map.idx or 1
+			indices[name] = ctf_map.get_idx_and_map(param)
 		end
 
+		indices[name] = indices[name] or (ctf_map.map and ctf_map.map.idx) or 1
+
 		if player then
-			show_catalog(name, idx or 1)
+			show_catalog(name)
 		else
 			minetest.chat_send_player(name, " *** CTF Map Catalog for IRC *** ")
 			if not param then
```

Comment:
```suggestion
			if not param then
				indices[name] = (ctf_map.map and ctf_map.map.idx) or 1
```

This is needed to really show the current map in IRC if no param is specified.

Reply:
I haven't removed it completely, I've just moved it further up the file.

Reply:
I just tested, and it seems to work perfectly alright, both in-game and via IRC.

Reply:
I think you are refering to (new) line 168.

However, this is not the same: In line 168, the last viewed map has first priority.
In IRC, we need to make sure that if no param is supplied, not the last viewed map is used, but the currently played map to ensure one can find out the current map via IRC. (/status is cut off often.) For this, you need to implement my suggestion above.

If you want to test this, use the following procedure for IRC:

1) `/maps` -> shows the currently played map :heavy_check_mark: 
2) `/maps <some other name>` -> shows the map <some other name> :heavy_check_mark: 
2) `/maps` -> shows <some other name>, although it should show the currently played map :x:

Reply:
Ah now I see what you mean. I thought showing the last viewed map was the correct behaviour ;)

Reply:
Do you want that to be the behaviour for the in-game maps catalog as well?

Reply:
Don't mind that question, it doesn't make sense to behave differently based on how the command is invoked.

Reply:
Actually, I'd prefer different behaviour in those different situations. If you are in-game, you usually know what the current map is or you can see it from the (un-truncated) /status output.

So I think the old behaviour should also be used in the new version, i.e. (if no param is specified):
 - In-Game: Show last viewed map (player probably wants to continue reading)
 - IRC: Show current map (last viewed map is in chat log anyways, player wants to know the current map)

Reply:
Done

Reply:
> Actually, I'd prefer different behaviour in those different situations. If you are in-game, you usually know what the current map is or you can see it from the (un-truncated) /status output.

That's needlessly complicated, IMO. But ok...